autoload/lsp/completion.vim

```diff
@@ -170,6 +172,25 @@ export def CompletionReply(lspserver: dict<any>, cItems: any)
     CompletionUltiSnips(prefix, items)
   endif
 
+  if opt.lspOptions.useBufferCompletion
+      # add completion from current buf
+      var words = {}
+      var text = join(getline(1, '$'), "\n")
+      for word in split(text, '\W\+')
```

Comment:
Instead of getting all the lines, joining them, and then splitting the words, it might be efficient to directly process the lines one at a time:
```
  for line in getline(1, '$')
    for word in line->split('\W\+')
```

Reply:
added fix in https://github.com/yegappan/lsp/pull/284/commits/b48c94b90e9da99d6fa271f0eba8400cb5d11d48